tools/cldr-code/src/main/java/org/unicode/cldr/util/BidiUtils.java

```diff
@@ -0,0 +1,162 @@
+    // more syntax
+
+    private enum SpanClass {
+        NUMBERS("\\p{N}"),
+        LETTERS_MARKS("[\\p{L}\\p{M}]"),
+        DATE_PUNCT("[+]"),
+        SPACES("\\p{Z}"),
+        OTHERS("\\p{any}") // must be last, to pick up remainder.
+    ;
+        final UnicodeSet uset;
+
+        private SpanClass(String unicodeSetSource) {
+            uset = new UnicodeSet(unicodeSetSource);
+        }
+
+        static {
+            // clean up by removing previous values
+            UnicodeSet soFar = new UnicodeSet();
+            for (SpanClass sc : SpanClass.values()) {
+                sc.uset.removeAll(soFar).freeze();
+                soFar.addAll(sc.uset);
+            }
+        }
+    }
+    /**
+     * Checks the ordering of the example, under the specified bidiDirectionOptions;
+     *
+     * @param example Source text, not HTMLified
+     * @param outputReorderedResults One string for each specified bidiDirectionOption
+     * @param bidiDirectionOptions an array of BIDI directions from com.ibm.icu.text.Bidi. if there
+     *     are no items, the default is DIRECTION_DEFAULT_LEFT_TO_RIGHT (dir="auto"),
+     *     DIRECTION_RIGHT_TO_LEFT (dir="rtl").
+     * @return true unless two or more of the resulting strings are different.
+     */
+    public static boolean isOrderingUnchanged(
+            String example, List<String> outputReorderedResults, int... bidiDirectionOptions) {
+        boolean hasList = outputReorderedResults != null;
+        if (!hasList) {
+            outputReorderedResults = new ArrayList<>();
+        } else {
+            outputReorderedResults.clear();
+        }
+        boolean result = true;
+        for (int count = 0; count < bidiDirectionOptions.length; ++count) {
+            String reordered = new Bidi(example, bidiDirectionOptions[count]).writeReordered(0);
+            outputReorderedResults.add(reordered);
+            if (result && count != 0 && !reordered.equals(outputReorderedResults.get(0))) {
+                result = false;
+                if (!hasList) {
+                    break; // if the output results are not needed, then stop.
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Return a list of the , where each span is a sequence of:
+     *
+     * @param orderedLTR
+     * @return
+     */
+    /**
+     * Gets the 'fields' in a formatted string, used to test whether bidi reordering causes the
+     * original fields to merge when reordered. Each field is the longest contiguous span of
+     * characters with the same properties: *
+     *
+     * <ul>
+     *   <li>numbers (\p{N})
+     *   <li>letters & marks ([\p{L}\p{M}
+     *   <li>Other
+     * </ul>
+     *
+     * @param ordered
+     * @return a set of fields, in the same order as found in the text but duplicates removed (ike
+     *     LinkedHashSeet).
+     */
+    public static Set<String> getFields(String reordred, Set<String> result) {
```

Comment:
`reordred` should be `reordered` and the comment above should have `@param reordered` and `@param result`